src/services/runtime/RuntimeMetadataService.ts

```diff
@@ -32,4 +34,45 @@ export class RuntimeMetadataService extends AbstractService {
 
 		return metadata;
 	}
+
+	/**
+	 * Fetch the requested version of `Metadata` in decoded JSON form.
+	 *
+	 * @param hash `BlockHash` to make call at
+	 */
+	async fetchMetadataVersioned(hash: BlockHash, metadataVersion: number): Promise<Metadata> {
+		const { api } = this;
+		const apiAt = await api.at(hash);
```

Comment:
Were potentially calling `api.at` twice, I would just pass the `api` from the controller down to the service and call it `apiAt` or something.